crates/shielded_token/src/masp/shielded_sync/utils.rs

```diff
@@ -8,16 +8,40 @@ use masp_primitives::sapling::{Node, Note, PaymentAddress, ViewingKey};
 use masp_primitives::transaction::Transaction;
 use namada_core::chain::BlockHeight;
 use namada_core::collections::HashMap;
+use namada_tx::event::MaspEventKind;
 use namada_tx::{IndexedTx, IndexedTxRange};
+use serde::{Deserialize, Serialize};
+
+/// An indexed masp tx carrying information on wether it was a fee paying tx or
+/// a normal transfer
+#[derive(
+    Debug,
+    Default,
+    Clone,
+    BorshSerialize,
+    BorshDeserialize,
+    PartialOrd,
+    PartialEq,
+    Eq,
+    Ord,
+    Serialize,
+    Deserialize,
+)]
+pub struct MaspIndexedTx {
+    /// The masp tx kind, fee-payment or transfer
+    pub kind: MaspEventKind,
```

Comment:
I know it feels like code duplication, but it might be better to have a separate type for the kind of masp indexed tx. we might want to introduce other masp event kinds that have nothing to do with the way we intend to sort masp indexed txs.

Reply:
Done in 4d9efbd24daa58662e04237dbe2d87528512b76